x-pack/legacy/plugins/maps/public/plugin.ts

```diff
@@ -21,11 +23,25 @@ export type MapsPluginStart = ReturnType<MapsPlugin['start']>;
 /** @internal */
 export class MapsPlugin implements Plugin<MapsPluginSetup, MapsPluginStart> {
   public setup(core: any, plugins: any) {
-    const app = plugins.__LEGACY.uiModules.get('app/maps', ['ngRoute', 'react']);
-    app.directive('mapListing', function(reactDirective: any) {
-      return reactDirective(wrapInI18nContext(MapListing));
-    });
+    const {
+      __LEGACY: { uiModules },
+      np: { licensing },
+    } = plugins;
+
+    uiModules
+      .get('app/maps', ['ngRoute', 'react'])
+      .directive('mapListing', function(reactDirective: any) {
+        return reactDirective(wrapInI18nContext(MapListing));
+      });
+
+    if (licensing) {
+      licensing.license$.subscribe(({ uid }: { uid: string }) => setLicenseId(uid));
+    }
   }
 
-  public start(core: CoreStart, plugins: any) {}
+  public start(core: CoreStart, plugins: any) {
+    initServicesAndConstants({
+      indexPatterns: plugins.np.data.indexPatterns,
```

Comment:
consider removing this indexPattern refactoring out of this PR. It's unrelated to the license-checking changes